kinode/src/kernel/process.rs

```diff
@@ -285,12 +285,17 @@ pub async fn make_process_loop(
                         .send(&send_to_terminal)
                         .await;
                 }
-                Err(_) => {
+                Err(e) => {
                     let stderr = wasi_stderr.contents().into();
                     let stderr = String::from_utf8(stderr)?;
+                    let output = if stderr != String::new() {
+                        stderr
+                    } else {
+                        format!("{}", e.root_cause())
+                    };
```

Comment:
Very nice.

Reply:
Made me realize we can use `.is_empty()` here in the conditional logic; 06c0bd4